nomad/structs/structs.go

```diff
@@ -2016,6 +2067,10 @@ func (n *NetworkResource) PortLabels() map[string]int {
 // Networks defined for a task on the Resources struct.
 type Networks []*NetworkResource
 
+// COMPAT(0.10): Remove in 0.10
+// ResourceDevices are part of Resources
+type ResourceDevices []*RequestedDevice
```

Comment:
It's the common pattern to define a type and follow with its methods immediately below.  I would suggest moving this definition closer to `ResourceDevices.Equals`.